### PR TITLE
Feature/#230 replay attack protection jwt

### PR DESCRIPTION
**Pull request by nataschaberg**
Implemenation of #230

### CODE REVIEW

File: tests/identityWallet/identityWallet.test.ts [hunk 1]
```diff
@@ -61,11 +57,12 @@ describe('IdentityWallet', () => {
...
     const sandbox = sinon.createSandbox()
     let stubCredCreate
-    let stubFromJWTEncodable
+    let spyFromJWTEncodable
+    let interactionToken
 
     before(() => {
       stubCredCreate = sandbox.stub(SignedCredential, 'create').callsFake(() => stubbedCredential)
-      stubFromJWTEncodable = sandbox.stub(JSONWebToken, 'fromJWTEncodable').callsFake(stubbedFromJWTEncodable)
+      spyFromJWTEncodable = sandbox.spy(JSONWebToken, 'fromJWTEncodable')
```
Comment (Exulansis) [inline tests/identityWallet/identityWallet.test.ts]: +1

Comment (Exulansis) [inline tests/identityWallet/identityWallet.test.ts]: +1

File: tests/interactionTokens/jsonWebToken.test.ts [hunk 2]
```diff
@@ -66,23 +71,31 @@ describe('JSONWebToken', () => {
...
     const jwt = new JSONWebToken()
-
+    const nonce = Math.random().toString(36)
+    
     jwt.setSignature(signature)
     jwt.setTokenContent(credReq)
     jwt.setTokenIssuer(iss)
+    jwt.setTokenTimeStamps()
+    jwt.setTokenNonce(nonce)
+    jwt.setTokenAudience('did:jolo:testauddid')
```
Comment (Exulansis) [inline tests/interactionTokens/jsonWebToken.test.ts]: Can we replace this with a did from the *.data.ts files?
Something like mockDID

File: ts/interactionTokens/JSONWebToken.ts [hunk 3]
```diff
@@ -117,7 +152,14 @@ export class JSONWebToken<T extends JWTEncodable> implements IDigestable {
   */
 
   public static decode<T extends JWTEncodable>(jwt: string): JSONWebToken<T> {
-    return JSONWebToken.fromJSON(decodeToken(jwt))
+    const jwtClass = JSONWebToken.fromJSON(decodeToken(jwt))
+    const validTimePeriod = jwtClass.getExpirationTime() - jwtClass.getIssueTime() > 1
```
Comment (Exulansis) [inline ts/interactionTokens/JSONWebToken.ts]: Can we instead do something like
```typescript
if (jwtClass.getExpirationDate() < Date.now()) {
  ...
}
```

I will add the change and check if it breaks tests.

File: tests/interactionTokens/jsonWebToken.test.ts (hunk 2 above)
Comment (nataschaberg) [reply tests/interactionTokens/jsonWebToken.test.ts]: 👍

File: ts/interactionTokens/JSONWebToken.ts (hunk 3 above)
Comment (nataschaberg) [reply ts/interactionTokens/JSONWebToken.ts]: good catch

**Pull request merged**